test/retained_topic_map.hpp

```diff
@@ -260,25 +269,26 @@ class retained_topic_map {
     }
 
     // Insert a value at the specified subscription path
-    size_t insert_or_update(MQTT_NS::string_view topic, Value const &value) {
+    template<typename V>
+    size_t insert_or_update(MQTT_NS::string_view topic, V&& value) {
         auto& direct_index = map.template get<direct_index_tag>();
         auto path = this->find_topic(topic);
 
         if (path.empty()) {
             auto new_topic = this->create_topic(topic);
-            direct_index.modify(new_topic, [&value](path_entry &entry){ entry.value.emplace(std::move(value)); });
+            direct_index.modify(new_topic, [value = fwd_capture(std::forward<V>(value))](path_entry &entry){ entry.value.emplace(access(value)); });
```

Comment:
I don't understand why you're using std::tuple to store the object in the lambda. Wouldn't 

```
[value = std::forward<V>(value)](...){...}
```

work just as well, without the extra complexity?

Reply:
Further, the call to `access()` here is receiving the `value` variable as an lvalue reference.

The lambda here is storing `value` by copy (or move, if value is an rvalue reference), so if the lambda is only ever called once, you can safely use std::move on the `entry.value.emplace(std::move(value))` function call, no need for the `access()` function or using tuples. But to do that, you'll need to make the lambda `mutable`.

If the lambda is called more than once, then you can't do that, but the way the code was written prior to this PR makes me think that the lambda is called once only.

Reply:
So it should be?

````C++
direct_index.modify(new_topic, [value = std::forward<V>(value)](path_entry &entry) mutable { entry.value.emplace(std::move(value)); });
````
Is that what you mean ?

Reply:
Ah yes, you can forward into lambda temporary, and lambda temporary moves into the entry.value